server.go

```diff
@@ -664,215 +538,161 @@ func (s *Server) handlerMovie(w http.ResponseWriter, r *http.Request) {
-	// internal error log
-	errors := []string{}
-	// bool to check if the site should be rerendered
-	rerenderSite := false
-	// slice for the api results
-	var results []string
+func (s *Server) handleAutofill(data *dataAddMovie, w http.ResponseWriter, r *http.Request) (results []string, links []string) {
+
+	// Get all needed values from the form
+
+	// Get all links from the corresponding input field
+	linktext := strings.ReplaceAll(r.FormValue("Links"), "\r", "")
+	data.ValLinks = linktext
+
+	// Get the remarks from the corresponding input field
+	remarkstext := strings.ReplaceAll(r.FormValue("Remarks"), "\r", "")
+	data.ValRemarks = remarkstext
+
+	// Check link maxlength
+	maxLinkLength, err := s.data.GetCfgInt(ConfigMaxLinkLength, DefaultMaxLinkLength)
+	if err != nil {
+		s.l.Error("Unable to get %q: %v", ConfigMaxLinkLength, err)
+		s.doError(
+			http.StatusInternalServerError,
+			"something went wrong :C",
+			w, r)
+		return
+	}
+
+	if len(linktext) > maxLinkLength {
+		s.l.Debug("Links too long: %d", len(linktext))
+		data.ErrorMessage = append(data.ErrorMessage, fmt.Sprintf("Links too long! Max Length: %d characters", maxLinkLength))
+		data.ErrLinks = true
+	}
+
+	// Check for valid links
+	links = strings.Split(linktext, "\n")
+	links, err = common.VerifyLinks(links)
+	if err != nil {
+		s.l.Error(err.Error())
+		data.ErrorMessage = append(data.ErrorMessage, "Invalid link(s) given.")
+		data.ErrLinks = true
+	}
+
+	var sourcelink string
 	// make sure we have a link to look at
-	if len(links) >= 1 {
-		sourcelink := links[0]
-		autofillEnabled, err := s.data.GetCfgBool(ConfigAutofillEnabled, DefaultAutofillEnabled)
+	if len(links) == 0 {
+		s.l.Error("no links given")
+		data.ErrorMessage = append(data.ErrorMessage, "No link found.")
+		data.ErrLinks = true
+	} else {
+		sourcelink = links[0]
+	}
+
+	// Check Remarks max length
+	maxRemarksLength, err := s.data.GetCfgInt(ConfigMaxRemarksLength, DefaultMaxRemarksLength)
+	if err != nil {
+		s.l.Error("Unable to get %q: %v", ConfigMaxRemarksLength, err)
+		s.doError(
+			http.StatusInternalServerError,
+			"something went wrong :C",
+			w, r)
+		return
+	}
+
+	if len(remarkstext) > maxRemarksLength {
+		s.l.Debug("Remarks too long: %d", len(remarkstext))
+		data.ErrorMessage = append(data.ErrorMessage, fmt.Sprintf("Remarks too long! Max Length: %d characters", maxRemarksLength))
+		data.ErrRemarks = true
+	}
+
+	// Exit early if any errors got reported
+	if data.isError() {
+		return nil, nil
+	}
+
+	if strings.Contains(sourcelink, "myanimelist") {
+		s.l.Debug("MAL link")
+
+		results, err = s.handleJikan(data, w, r, sourcelink)
+
 		if err != nil {
-			s.l.Error("Error while retriving config value %s:\n %v", ConfigAutofillEnabled, err)
-			errors = append(errors, "Something went wrong :C")
-			rerenderSite = true
-			return nil, errors, rerenderSite
+			return nil, nil
```

Comment:
The error returned from handleJikan should be logged here.

Reply:
Will be changed in the next commit